tests/integration/git_annex_archive_test.go

```diff
@@ -0,0 +1,197 @@
+package integration
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"testing"
+
+	auth_model "code.gitea.io/gitea/models/auth"
+	"code.gitea.io/gitea/modules/annex"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/setting"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGitAnnexArchive(t *testing.T) {
+	onGiteaRun(t, func(t *testing.T, u *url.URL) {
+		ctx := NewAPITestContext(t, "user2", "annex-archive-test", auth_model.AccessTokenScopeWriteRepository)
+		require.NoError(t, doCreateRemoteAnnexRepository(t, u, ctx, false))
+		req := NewRequest(t, "GET", u.String())
+		_ = ctx.Session.MakeRequest(t, req, http.StatusOK)
+
+		// cleanup previously generated archives
+		adminSession := loginUser(t, "user1")
+		adminToken := getTokenForLoggedInUser(t, adminSession, auth_model.AccessTokenScopeWriteAdmin)
+		link, _ := url.Parse("/api/v1/admin/cron/delete_repo_archives")
+		link.RawQuery = url.Values{"token": {adminToken}}.Encode()
+		resp := adminSession.MakeRequest(t, NewRequest(t, "POST", link.String()), http.StatusNoContent)
+		bs, err := io.ReadAll(resp.Body)
+		require.NoError(t, err)
+		require.Len(t, bs, 0)
+
+		remoteRepoPath := path.Join(setting.RepoRootPath, ctx.GitPath())
+
+		// get the commitID of the master branch
+		repo, err := git.OpenRepository(git.DefaultContext, remoteRepoPath)
+		require.NoError(t, err)
+		commitID, err := repo.GetBranchCommitID("master")
+		require.NoError(t, err)
+		tree, err := repo.GetTree("master")
+		require.NoError(t, err)
+		entries, err := tree.ListEntriesRecursiveFast()
+		require.NoError(t, err)
+		filesInGit := make([]string, 0, len(entries))
+		for _, entry := range entries {
+			if !entry.IsDir() {
+				filesInGit = append(filesInGit, entry.Name())
+			}
+		}
+
+		t.Run("TARGZ", func(t *testing.T) {
+			// request a tar.gz archive of the repo
+			link, _ := url.Parse(fmt.Sprintf("/api/v1/repos/%s/%s/archive/master.tar.gz", ctx.Username, ctx.Reponame))
+			resp := ctx.Session.MakeRequest(t, NewRequest(t, "GET", link.String()), http.StatusOK)
+			bs, err := io.ReadAll(resp.Body)
+			require.NoError(t, err)
+
+			// open the archive for reading
+			gzrd, err := gzip.NewReader(bytes.NewReader(bs))
+			require.NoError(t, err)
+			defer gzrd.Close()
+			rd := tar.NewReader(gzrd)
+
+			var filesInArchive []string
+			for {
+				header, err := rd.Next()
+				if err == io.EOF {
+					break
+				}
+				require.NoError(t, err)
+
+				// skip directories
+				if header.Typeflag == tar.TypeDir {
+					continue
+				}
+
+				// check that the pax_global_header comment field is correctly set
+				if path.Base(header.Name) == "pax_global_header" {
+					require.Equal(t, commitID, header.PAXRecords["comment"])
```

Comment:
I didn't know about [PAX headers](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/pax.html#tag_20_92_13_03)!

Reply:
I also didn't, at first. I think I found out about that when reading the `git archive` generated tar file with go for the first time and encountering that file name. What makes matters worse: it is very much not straightforward or even impossible to read these comments out using gnu tar. At least I did not find the option for that, yet.